src/main/java/com/google/api/codegen/metadatagen/PackageMetadataContext.java

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.metadatagen;
+
+import com.google.api.codegen.CodegenContext;
+import com.google.api.tools.framework.snippet.Doc;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import org.yaml.snakeyaml.Yaml;
+
+/*
+ * A Context object to be passed into the snippet template rendering phase of package metadata
+ * generation.
+ */
+public class PackageMetadataContext extends CodegenContext {
```

Comment:
I recommend against re-using CodegenContext. It is the root of the unmanageable pre-MVVM context hierarchy.